docs/advanced/tls_hardware_acceleration.md

```diff
@@ -0,0 +1,131 @@
+
+### What parts can I accelerate?
+
+mbed TLS has separate modules for the different cryptographic primitives. Hardware acceleration interface is available for the following modules and functions:
+
+- Symmetric
+    - AES: `mbedtls_aes_setkey_enc()`, `mbedtls_aes_setkey_dec()`, `mbedtls_aes_encrypt()`, `mbedtls_aes_decrypt()`
+    - ARC4
+    - BLOWFISH
+    - CAMELLIA
+    - DES: `mbedtls_des_setkey()`, `mbedtls_des_crypt_ecb()`, `mbedtls_des3_crypt_ecb()`
+    - XTEA
+    - MD2: `mbedtls_md2_process()`
+    - MD4: `mbedtls_md4_process()`
+    - MD5: `mbedtls_md5_process()`
+    - RIPEMD160: `mbedtls_ripemd160_process()`
+    - SHA1: `mbedtls_sha1_process()`
+    - SHA256: `mbedtls_sha256_process()`
+    - SHA512: `mbedtls_sha512_process()`
+- Asymmetric:
+    - ECP: `mbedtls_internal_ecp_randomize_jac()`, `mbedtls_internal_ecp_add_mixed()`, `mbedtls_internal_ecp_double_jac()`, `mbedtls_internal_ecp_normalize_jac_many()`, `mbedtls_internal_ecp_normalize_jac()`, `mbedtls_internal_ecp_double_add_mxz()`, `mbedtls_internal_ecp_randomize_mxz()`, `mbedtls_internal_ecp_normalize_mxz()`
+
+<span class="notes">**Note:** The ECP module and function replacement is not part of mbed OS yet. It is in an Early Partner Preview stage and it is available for evaluation in a [PR on github](https://github.com/ARMmbed/mbed-os/pull/3294)</span>
+
+You can extend functionality by either overriding certain functions or replacing the whole module. The easier and safer way of doing this is to override some or all of the functions in a particular module. Sometimes this won't be enough, usually because of a need to change the data structures or the higher level algorithms. If this is the case, you'll need to replace the whole module. Please note that in the case of ECP functions the override is only partial, mbed TLS will fall back to the software implementation if the hardware cannot handle a particular group.
+
+### How can I make mbed TLS use my hardware accelerator?
+
+You have to provide an alternative implementation for the parts of mbed TLS that you want to accelerate.
+
+mbed TLS has a variety of options to make use of your alternative implementation. These make it possible to easily replace functionality at various abstraction levels and to different extents. In other words, you can replace the least amount of code to reach the highest possible acceleration with the smallest amount of effort.
+
+## Adding acceleration by replacing functions
+
+First, you should consider what kind of functionality your hardware provides. Does the processor have some modular arithmetic capabilities? Or does your board have a full cryptographic module, securely storing keys and providing the functionality of high level cryptographic primitives?
```

Comment:
This section has a bias towards asymmetric acceleration, but in reality most acceleration will come from AES and SHA. Can we change the bias a little to help the most common users?